### PR TITLE
Use output size helpers for cryptographic primitives

### LINKED ISSUE #67059: Estimated signature size calculation for ECDsa and RSA may be incorrect
Based on recent discussions with @bartonjs where I needed to also obtain an estimated size. As per Jeremy, ECDsa should use `2 * ((KeySize + 7) / 8)` and for RSA `(KeySize + 7) / 8`.

What is currently being used is this:
https://github.com/dotnet/runtime/blob/ea4ebaa3c5162bcabc63284ba3b59aa683912af4/src/libraries/Common/src/System/Security/Cryptography/ECDsaCng.SignVerify.cs#L21-L29

and this:
https://github.com/dotnet/runtime/blob/ea4ebaa3c5162bcabc63284ba3b59aa683912af4/src/libraries/Common/src/System/Security/Cryptography/RSACng.SignVerify.cs#L62

For ECDsa and RSA respectively. 

**It is not a bug** as the code handles the case where the signature buffer wasn't big enough but I think we should be consistent on how to calculate it, maybe even consider adding a `GetSignatureSize()` API.

### PR DESCRIPTION
**Pull request by vcsjones**
This updates a few places that can use algorithm output size helpers.

Closes #67059

### CODE REVIEW

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @vcsjones !

One day we need to discuss our benchmarks coverage:

https://github.com/dotnet/performance/tree/main/src/benchmarks/micro/libraries/System.Security.Cryptography.Primitives
https://github.com/dotnet/performance/tree/main/src/benchmarks/micro/libraries/System.Security.Cryptography.X509Certificates
https://github.com/dotnet/performance/tree/main/src/benchmarks/micro/libraries/System.Security.Cryptography

File: src/libraries/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs [hunk 1]
```diff
@@ -198,7 +198,7 @@ private unsafe bool TryEncryptOrDecrypt(ReadOnlySpan<byte> data, Span<byte> dest
         // Now that the padding mode and information have been marshaled to their native counterparts, perform the encryption or decryption.
         private unsafe byte[] EncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan<byte> input, AsymmetricPaddingMode paddingMode, void* paddingInfo, bool encrypt)
         {
-            int estimatedSize = KeySize / 8;
+            int estimatedSize = GetMaxOutputSize();
```
Comment (stephentoub) [inline src/libraries/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:201]: This had been doing `KeySize / 8` and is now doing `(KeySize + 7) >>> 3`.  I assume this is fine, as this is an estimate and ceiling is better than floor.  But just checking...

File: src/libraries/Common/src/System/Security/Cryptography/ECDsaCng.SignVerify.cs [hunk 2]
```diff
@@ -20,15 +20,7 @@ public override byte[] SignHash(byte[] hash)
...
-            int estimatedSize = KeySize switch
-            {
-                256 => 64,
-                384 => 96,
-                521 => 132,
-                // If we got here, the range of legal key sizes for ECDsaCng was expanded and someone didn't update this switch.
-                // Since it isn't a fatal error to miscalculate the estimatedSize, don't throw an exception. Just truck along.
-                _ => KeySize / 4,
-            };
+            int estimatedSize = GetMaxSignatureSize(DSASignatureFormat.IeeeP1363FixedFieldConcatenation);
```
Comment (stephentoub) [inline src/libraries/Common/src/System/Security/Cryptography/ECDsaCng.SignVerify.cs:23]: Is this more expensive? The implementation of the method isn't trivial, but maybe it doesn't matter?

File: src/libraries/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs (hunk 1 above)
Comment (vcsjones) [reply src/libraries/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:201]: Yeah. Straight "divide by 8" was technically wrong but worked in practice. The ceiling calculation is "right".

File: src/libraries/Common/src/System/Security/Cryptography/ECDsaCng.SignVerify.cs (hunk 2 above)
Comment (vcsjones) [reply src/libraries/Common/src/System/Security/Cryptography/ECDsaCng.SignVerify.cs:23]: Maybe in a micro-benchmark, but ECDSA signing and verifying is going to dominate. For `IeeeP1363FixedFieldConcatenation` it basically boils down to `AsymmetricAlgorithmHelpers.BitsToBytes(KeySize) * 2`; Where `BitsToBytes` just does the "ceiling then divide by 8". I wouldn't be surprised if the JIT inlined `BitsToBytes`.

**Review by bartonjs — APPROVED** [review]

**Review by jozkee — APPROVED** [review]

**Comment by Jozkee** [comment]
System.Net.Http.Json failure is https://github.com/dotnet/runtime/issues/82918.
System.Net[.Http] failures are most likely instability. 
System.Runtime.Tests [failure ](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-82800-merge-9a343e60e8e147a796/System.Runtime.Tests.Attempt.3/1/console.5d9b6836.log?helixlogtype=result)on MacCatalystc says:
```
[15:00:42] fail: Application test run crashed
                 No test log file was produced
```

**Pull request merged**